### ISSUE TITLE
bro can u help me with funcaptcha twitch callback function

### ISSUE THREAD

**Issue by Its-Zeus** [eperezcosano/2captcha-twitch#1]
help me :)

**Comment by eperezcosano** [comment]
Hey!
I've struggling getting the callback function for the twitch funcaptcha as it is inside of a few nested iframes it is hard to debug but I finally got it.

With devtools you can debug all event messages the iframes exchange with
`window.addEventListener('message', console.log)`
for each iframe. (Select each iframe inside de console tab like so https://firefox-source-docs.mozilla.org/devtools-user/working_with_iframes/index.html -chrome is similar-)

Then when the captcha is completed two events are fired:
![image](https://user-images.githubusercontent.com/7610219/168116282-f2f4dce5-6add-42ca-9062-e3b29bab757f.png)

So theoretically the solution is executing your custom events like:
`window.postMessage('complete', '*')`
and
`window.postMessage('{\"token\":\"842627d2833c42934.3221872805|r=eu-west-1|metabgclr=transparent|guitextcolor=%23000000|metaiconclr=%23757575|meta=3|lang=en|pk=9260643B-49E2-EF5B-2CB5-30FD3451DF51|at=40|atp=2|cdn_url=https%3A%2F%2Ftwitch-api.arkoselabs.com%2Fcdn%2Ffc|lurl=https%3A%2F%2Faudio-eu-west-1.arkoselabs.com|surl=https%3A%2F%2Ftwitch-api.arkoselabs.com\",\"data\":{\"token\":\"842627d2833c42934.3221872805|r=eu-west-1|metabgclr=transparent|guitextcolor=%23000000|metaiconclr=%23757575|meta=3|lang=en|pk=9260643B-49E2-EF5B-2CB5-30FD3451DF51|at=40|atp=2|cdn_url=https%3A%2F%2Ftwitch-api.arkoselabs.com%2Fcdn%2Ffc|lurl=https%3A%2F%2Faudio-eu-west-1.arkoselabs.com|surl=https%3A%2F%2Ftwitch-api.arkoselabs.com\"},\"key\":\"9260643B-49E2-EF5B-2CB5-30FD3451DF51\",\"message\":\"challenge completed\",\"type\":\"broadcast\"}", origin: "https://twitch-api.arkoselabs.com', '*')`

However it is not enough because it seems the backend is performing more comprobations afterwards. I pressume it checks if the token offered at the start for this user is the same as the submitted one. Therefore, I think is impossible to bypass the captcha with the token method.

In conclusion, my workaround was using the grid method and click one by one each step.

If someone finds the answer for the token method I would be grateful if you share it.

**Comment by Its-Zeus** [comment]
Hi bro,
Thank you so much for your time I really appreciate your help.
Actually I figured out how to get the callback function after hours and hours and its working on the console
document.querySelector('div.Layout-sc-nxg1ff-0.gZaqky').children[0].__reactEventHandlers$b1y53owi77m.children.props.onChange('here the token')
but the problem is this react element : .__reactEventHandlers$b1y53owi77m is variable
the $b1y53owi77m change every time 
I wonder if I can get the value of it using selenium python
https://i.ibb.co/cQyBkgq/ty.png
And thank you again

**Comment by eperezcosano** [comment]
You can use simple javascript to get the element:
`
const obj = document.querySelector('div.Layout-sc-nxg1ff-0.gZaqky').children[0]');
`
`
const eventHandler = Object.entries(obj).find(pair => pair[0].startsWith('__reactEventHandlers'))[0] 
`
`
const callback = obj[eventHandler].children.props.onChange
`
Then you can pass the token to the callback. Tell me if it works.
Also, try with my other solution of "window.postMessage('complete', '*')"

**Comment by Its-Zeus** [comment]
I tried this one"window.postMessage('complete', '*')"  but it didnt work.
The other one its sending the captcha call like the captcha is passed but I get this error: That Captcha did not work. Please try again.
Please try again.
The same error I was getting when I tried to create twitch account using post request + 2captcha
its like they detect that the IP that solved the captcha is not the one creating the account I THINK. or maybe the client id
https://i.ibb.co/QD5Cr9c/eez.png

**Comment by Its-Zeus** [comment]
can u give me your telegram or discord ... ?

**Comment by eperezcosano** [comment]
Telegram @ethanpc
